modules/openapi-generator/src/main/resources/rust/request.rs

```diff
@@ -0,0 +1,238 @@
+            uri_str += "?";
+            uri_str += &query_string_str;
+        }
+        let uri: hyper::Uri = match uri_str.parse() {
+            Err(e) => {
+                return Box::new(futures::future::err(Error::UriError(e)));
+            }
+            Ok(u) => u,
+        };
+
+        let mut req = hyper::Request::new(self.method, uri);
+        {
+            let req_headers = req.headers_mut();
+            if let Some(ref user_agent) = conf.user_agent {
+                req_headers.set(UserAgent::new(Cow::Owned(user_agent.clone())));
+            }
+
+            req_headers.extend(headers.iter());
+
+            for (key, val) in raw_headers {
+                req_headers.set_raw(key, val);
+            }
+        }
+
+        if self.form_params.len() > 0 {
+            req.headers_mut().set(hyper::header::ContentType::form_url_encoded());
+            let mut enc = ::url::form_urlencoded::Serializer::new("".to_owned());
+            for (k, v) in self.form_params {
+                enc.append_pair(&k, &v);
+            }
+            req.set_body(enc.finish());
+        }
+
+        if let Some(body) = self.serialized_body {
+            req.headers_mut().set(hyper::header::ContentType::json());
+            req.headers_mut()
+                .set(hyper::header::ContentLength(body.len() as u64));
+            req.set_body(body);
+        }
+
+        let no_ret_type = self.no_return_type;
+        let res = conf.client
+                .request(req)
+                .map_err(|e| Error::from(e))
+                .and_then(|resp| {
+                    let status = resp.status();
+                    resp.body()
+                        .concat2()
+                        .and_then(move |body| Ok((status, body)))
+                        .map_err(|e| Error::from(e))
+                })
+                .and_then(|(status, body)| {
+                    if status.is_success() {
+                        Ok(body)
+                    } else {
+                        Err(Error::from((status, &*body)))
+                    }
+                });
+        Box::new(
+            res
+                .and_then(move |body| {
+                    let parsed: Result<U, _> = if no_ret_type {
+                        // This is a hack; if there's no_ret_type, U is (), but serde_json gives an
+                        // error when deserializing "" into (), so deserialize 'null' into it
+                        // instead.
+                        // An alternate option would be to require U: Default, and then return
+                        // U::default() here instead since () implements that, but then we'd
+                        // need to impl default for all models.
+                        serde_json::from_str("null")
+                    } else {
+                        serde_json::from_slice(&body)
+                    };
+                    parsed.map_err(|e| Error::from(e))
+                })
+        )
+    }
```

Comment:
More or less.

I did add minor error handling around uri parsing that was an unwrap before and restructured it a bit to make it read cleaner, but it's meant to be just a straight copy+paste.

Other than forms. I did add form parameters since those weren't present at all before.